### PR TITLE
Drop requirement for a CLA

### PR DESCRIPTION
**Pull request by danh-arm**
The Mbed Crypto project no longer requires a CLA. Contributions from now on
must be made under both Apache-2.0 AND GPL-2.0-or-later licenses, to enable
LTS (Long Term Support) branches of the software to continue to be provided
under either Apache-2.0 OR GPL-2.0-or-later. Contributors must accept the terms of the Developer Certificate of Origin (DCO) by adding a Signed-off-by: line to each commit
message.

The software on the development branch continues to be provided under
Apache-2.0.

Update README.md and CONTRIBUTING.md to explain the new licensing model.
Add a copy of the DCO to the project.

Expand the full Apache-2.0 license text in the LICENSE file and remove the
redundant apache-2.0.txt.

Fix inconsistent list formatting in CONTRIBUTING.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 The Mbed cryptography library is a reference implementation of the cryptography interface of the Arm Platform Security Architecture (PSA). This is a preview release of Mbed Crypto, provided for evaluation purposes only.
 
-Mbed Crypto is distributed under the Apache License, version 2.0. See the [LICENSE](LICENSE) file for the full text of the license.
+## License
+
+Unless specifically indicated otherwise in a file, Mbed TLS files are provided under the [Apache-2.0](https://spdx.org/licenses/Apache-2.0.html) license. See the [LICENSE](LICENSE) file for the full text of this license. Contributors must accept that their contributions are made under both the Apache-2.0 AND [GPL-2.0-or-later](https://spdx.org/licenses/GPL-2.0-or-later.html) licenses. This enables LTS (Long Term Support) branches of the software to be provided under either the Apache-2.0 OR GPL-2.0-or-later licenses.
```
Comment (yanesca) [inline README.md]: As discussed earlier, we could try clarifying the inbound licenses a little bit. Maybe writing something like this in addition: "The inbound licences are both Apache-2.0 AND GPL-2.0-or-later simultaneously, that is, contributions are made under both."

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,7 +19,8 @@ Making a Contribution
 1. Write a test which shows that the bug was fixed or that the feature works as expected.
 1. Send a pull request (PR) and work with us until it gets merged and published. Contributions may need some modifications, so a few rounds of review and fixing may be necessary. We will include your name in the ChangeLog :)
 1. For quick merging, the contribution should be short, and concentrated on a single feature or topic. The larger the contribution is, the longer it would take to review it and merge it.
-1. Mbed TLS is released under the Apache license, and as such, all the added files should include the Apache license header.
+1. Ensure that all new files include the [Apache-2.0](https://spdx.org/licenses/Apache-2.0.html) standard license header.
+1. Ensure that each commit has at least one `Signed-off-by:` line, using your real name and email address. The names in the `Signed-off-by:` and `Author:` lines must match. If anyone else contributes to the commit, they must also add their own `Signed-off-by:` line. By adding this line the contributor(s) certify that the contribution is made under the terms of the [Developer Certificate of Origin](dco.txt).
```
Comment (yanesca) [inline CONTRIBUTING.md]: Additionally to the dco, we could consider mentioning the inbound licences here too. (The dco makes a vague reference to an open source licence, it is not impossible to figure out what it is, but I think we should make the job easier for our contributors.)

Comment (danh-arm) [reply CONTRIBUTING.md]: OK, I'll link the license section of the readme here.

File: README.md (hunk 1 above)
Comment (danh-arm) [reply README.md]: I think I already clarified that point. Your suggestion just says the same thing twice IMO.

**Comment by mpg** [comment]
Note: the CI is only failing the Mbed OS tests, which are a known issue clearly independent from this PR.

**Comment by yanesca** [comment]
The CI fails on Mbed OS, but not for the known issue we had. This PR deletes the file `apache-2.0.txt`, which is targeted explicitly by the importer script in Mbed OS.

@Patater The Mbed OS importer script will need updating once this PR is merged.

**Comment by Patater** [comment]
> The CI fails on Mbed OS, but not for the known issue we had. This PR deletes the file `apache-2.0.txt`, which is targeted explicitly by the importer script in Mbed OS.
> 
> @Patater The Mbed OS importer script will need updating once this PR is merged.

Thanks for the heads up. We'll fix this when we update the version of Mbed TLS in Mbed OS.

**Pull request merged**